pkg/apis/serving/v1alpha1/llm_inference_service_types.go

```diff
@@ -0,0 +1,262 @@
+/*
+Copyright 2025 The KServe Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	igwapi "sigs.k8s.io/gateway-api-inference-extension/api/v1alpha2"
+	gatewayapi "sigs.k8s.io/gateway-api/apis/v1"
+)
+
+// LLMInferenceService is the Schema for the llminferenceservices API.
+// +k8s:openapi-gen=true
+// +kubebuilder:object:root=true
+type LLMInferenceService struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   LLMInferenceServiceSpec   `json:"spec,omitempty"`
+	Status LLMInferenceServiceStatus `json:"status,omitempty"`
+}
+
+// LLMInferenceServiceConfig is the Schema for the llminferenceserviceconfigs API.
+// +k8s:openapi-gen=true
+// +kubebuilder:object:root=true
+type LLMInferenceServiceConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   LLMInferenceServiceSpec         `json:"spec,omitempty"`
+	Status LLMInferenceServiceConfigStatus `json:"status,omitempty"`
+}
+
+// LLMInferenceServiceSpec defines the desired state of LLMInferenceService
+type LLMInferenceServiceSpec struct {
+	// Type of the deployment, e.g., 'default' or 'llm-d'. 'default' uses Raw deployments.
+	Type string `json:"type"`
+
+	// Model specification.
+	Model ModelSpec `json:"model"`
+
+	// Template for the main pod spec. In a multi-node case, this configures the "head" pod.
+	// +optional
+	Template *corev1.PodSpec `json:"template,omitempty"`
+
+	// WorkloadSpec configurations of the inference deployment.
+	// +optional
+	*WorkloadSpec `json:",inline,omitempty"`
+
+	// Parallelism configurations for the runtime.
+	// +optional
+	Parallelism *ParallelismSpec `json:"parallelism,omitempty"`
+
+	// Router configuration for the service.
+	// +optional
+	Router *RouterSpec `json:"router,omitempty"`
+
+	// Prefill configuration for disaggregated serving.
+	// +optional
+	Prefill *WorkloadSpec `json:"prefill,omitempty"`
```

Comment:
It looks like Prefill also needs to have parallelism, but currently there’s no dedicated `WorkloadSpec` for it.
Do you think it makes sense to create a separate struct for Prefill that includes both `WorkloadSpec` and `ParallelismSpec`?

Since `WorkloadSpec` is also used for multi-node workers, and those don’t need parallelism, it feels cleaner to separate them.

Reply:
I've added `Parallelism *ParallelismSpec` to `WorkloadSpec`